pkg/vmprovider/providers/vsphere/constants/constants.go

```diff
@@ -72,6 +72,9 @@ const (
 	CloudInitGuestInfoUserdata         = "guestinfo.userdata"
 	CloudInitGuestInfoUserdataEncoding = "guestinfo.userdata.encoding"
 
+	// CloudInitGuestInfoMaxSize the maximum allowed size for cloud init metadata.
+	CloudInitGuestInfoMaxSize = 64 * 1024
```

Comment:
Curious: is there a way to fetch this at a cluster level?  That way, we don't have to maintain this bespoke const in our code and worry about it falling out of sync with what is enforced at the Guest level.

If not, we should at least call out in comments that this is enforced by the infra (ESXi) (and maybe link a public KB) so readers are not wondering how we arrived at this const.